### ISSUE TITLE
[react/destructuring-assignment] Shows weird behaviour when I declare a variable with the name context

### ISSUE THREAD

**Issue by roman-16** [jsx-eslint/eslint-plugin-react#2309]
In a functional component I get an error when I declare a variable with the name context.
Example code:

```javascript
const Test = props => {
  const context = useContext(someContext);

  return (
    <div {...props}>
      {context.test} // Here message: 'Must use destructuring context'
    </div>
  );
}
```

**Comment by ljharb** [comment]
The error here should appear imo with any variable name coming from the useContext hook. Does the error disappear if you use a different variable name?

**Comment by wa4-fearless-otter** [comment]
No, then it doesn't appear.

**Comment by ljharb** [comment]
There seems like two bugs then:
1. the variable name `context` shouldn't be special
2. the return of `useContext` should be caught by this rule with every var name (altho this might need an option)

**Comment by Zinyon** [comment]
I think the error message on that case is correct following this [rule,](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/destructuring-assignment.md) isn't it?
can you explain more why there is a bug ?

**Comment by ljharb** [comment]
@Zinyon the rule is supposed to enforce _legacy_ context being destructured; it's not supposed to warn on `useContext`, or on any variable that happens to be named `context`.

**Comment by Zinyon** [comment]
@ljharb Thank you, it makes sense! and I'll try fix this.
